browser/autocomplete/sources.gni

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2021 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import("//brave/components/brave_webtorrent/browser/buildflags/buildflags.gni")
+import("//brave/components/ipfs/buildflags/buildflags.gni")
+import("//extensions/buildflags/buildflags.gni")
+
+brave_browser_autocomplete_sources = [
+  "//brave/browser/autocomplete/brave_autocomplete_scheme_classifier.cc",
+  "//brave/browser/autocomplete/brave_autocomplete_scheme_classifier.h",
+]
+
+brave_browser_autocomplete_deps = [
+  "//base",
```

Comment:
some of `sources.gni` has this dependency, while others are not - I'm guessing we can actually omit it

Reply:
In this particular case it's because of the `#include "base/strings/string_util.h"` [include from brave/browser/autocomplete/brave_autocomplete_scheme_classifier.cc](https://github.com/brave/brave-core/blob/master/browser/autocomplete/brave_autocomplete_scheme_classifier.cc#L10).

As for the inconsistency with other files, I think I might have made a mistake and didn't notice because the gn_check step didn't catch it now that this has been assimilated into the main //chrome/browser target. I'll do another pass over all these new sources.gni file and will add what's missing (vs removing them, as I think it's more interesting, see my comment in https://github.com/brave/brave-core/pull/8538#discussion_r617560485)

Reply:
Done another pass and fix a few missing deps (sorry not having noticed earlier), put the difference as fixup commits on top of this PR to ease review (will squash them before merge).